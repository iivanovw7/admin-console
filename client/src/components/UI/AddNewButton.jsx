import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import withWidth from '@material-ui/core/withWidth';

const AddNewButton = props => {

  const { classes, width, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  const mobileButton = () => {
    return (
      <Button size="small" className={classes.margin}
              onClick={() => {
                handleClick();
              }}>
        NEW
      </Button>
    );
  };

  const desktopButton = () => {
    return (
      <Fab color="primary" size="small" aria-label="Add" className={classes.fab}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    );
  };

  const components = {
    xs: mobileButton(),
    sm: mobileButton()
  };

  const Component = components[width] || desktopButton();

  return (
    <div className={classes.formRoot}>
      {Component}
    </div>
  );

};

AddNewButton.propTypes = {
  width: PropTypes.string.isRequired
};

export default withWidth()(AddNewButton);

