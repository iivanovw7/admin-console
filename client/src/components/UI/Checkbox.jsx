import Checkbox from '@material-ui/core/Checkbox';
import PropTypes from 'prop-types';
import React from 'react';

export const ActionCheckBox = props => {

  const { status, classes, handleClick } = props;

  return (
    <Checkbox
      className={classes.actionCheckBox}
      checked={status}
      size={'small'}
      color={'primary'}
      onChange={handleClick}
      disabled={false}
    />
  );

};

ActionCheckBox.propTypes = {
  classes: PropTypes.object,
  handleClick: PropTypes.func,
  status: PropTypes.bool
};
