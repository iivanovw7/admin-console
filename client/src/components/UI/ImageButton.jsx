import { Button, IconButton, ListItem } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const ImageButton = props => {

  const { classes, image, alt, handleClick, title } = props;

  return (
    <ListItem style={{ padding: 0, marginTop: 3, marginBottom: 3 }}>
      <IconButton className={classes.desktop} color="default" aria-label={alt} onClick={handleClick}
                  size={'small'}>
        <img style={{ width: '24px' }} src={image} alt={alt}/>
      </IconButton>
      <Button
        style={{ fontSize: '0.5rem' }}
        className={classes.mobile} size="small"
        onClick={handleClick}
      >
        {title}
      </Button>
    </ListItem>
  );
};

ImageButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  image: PropTypes.any.isRequired
};

