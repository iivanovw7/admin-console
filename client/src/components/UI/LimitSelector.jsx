import React from 'react';
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

export const LimitSelector = props => {

  const { classes, limit, handleLimit, limits } = props;

  return (
    <div className={classes.fromLimitSelector}>
      <FormControl className={classes.formControl}>
        <InputLabel>Results limit</InputLabel>
        <Select
          native
          value={limit}
          onChange={handleLimit}
        >
          {limits.map(newLimit => (
            <option key={newLimit} value={newLimit}>
              {newLimit}
            </option>
          ))}
        </Select>
      </FormControl>
    </div>
  );

};

