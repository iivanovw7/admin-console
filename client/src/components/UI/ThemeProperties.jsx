import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

