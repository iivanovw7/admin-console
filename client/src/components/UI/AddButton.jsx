import { withStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import { Buttons } from './ThemeProperties';

const AddButton = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button size="small" className={classes.mobile}
              onClick={() => {
                handleClick();
              }}>
        <strong>Create</strong>
      </Button>
      <Fab color="primary" size="small" aria-label="Add" className={classes.desktop}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

AddButton.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(AddButton);