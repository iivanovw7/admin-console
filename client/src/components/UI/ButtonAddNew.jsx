import { Button, Fab } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import PropTypes from 'prop-types';
import React from 'react';
import { Buttons } from './ThemeProperties';

const ButtonAddNew = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button
        size="small"
        className={classes.mobile}
        onClick={() => {
          handleClick();
        }}
      >
        <strong>Create</strong>
      </Button>
      <Fab
        color="primary"
        size="small"
        aria-label="Add"
        className={classes.desktop}
        onClick={() => {
          handleClick();
        }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

ButtonAddNew.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(ButtonAddNew);