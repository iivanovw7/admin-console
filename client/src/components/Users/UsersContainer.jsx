import { Link, Paper, Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {
  getBranches,
  getGroups,
  getRoles,
  getSingleUser,
  getUsers,
  searchUsers
} from '../../actions/index';
import AlertSnackbar from '../UI/Notifications/Snackbar';
import { Container } from '../UI/ThemeProperties';
import { actionCheckBox } from './UsersActionCheckBox';

const UsersContainer = props => {
  const { classes, history, dispatch, users, limit, page, search } = props;
  const currPage = page || 1;
  const currLimit = limit || 8;

  function displayStatus(status) {
    return status ? 'Active' : 'Disabled';
  }

  function handleUserClick(id) {
    dispatch(getSingleUser(id, history));
    dispatch(getRoles(1, 100, history));
    dispatch(getGroups(1, 100, history));
    dispatch(getBranches(1, 100, history));
  }

  //Triggers notification if there are any messages in props
  function displayNotifications() {
    if (props.errorMessage && !props.messageConfirmed) {
      return displayNotification(props.errorMessage, false);
    }
    if (props.successMessage && !props.messageConfirmed) {
      return displayNotification(props.successMessage, true);
    }
  }

  //Returns notification container
  const displayNotification = (message, success) => (

    <AlertSnackbar
      dispatch={dispatch}
      message={message}
      afterConfirm={() => {
        search ?
          dispatch(searchUsers(currPage, currLimit, search, history)) :
          dispatch(getUsers(currPage, currLimit, history));
      }}
      success={success}
    />
  );

  return (
    <Paper className={classes.root}>
      <Table className={classes.tables}>
        <TableHead>
          <TableRow>
            <TableCell className={classes.userNameCell}>
              <h3>Name</h3>
            </TableCell>
            <TableCell className={classes.userEmailCell} align="center">
              <h3>Email</h3>
            </TableCell>
            <TableCell className={classes.userStatusCell} align="center">
              <h3>Status</h3>
            </TableCell>
            <TableCell className={classes.userControlCell} align="center">
              <h3>Deactivate</h3>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {users.map(row => (
            <TableRow key={row._id}>
              <TableCell component="th" scope="row" className={classes.userNameCell}>
                <Link
                  component="button"
                  variant="body2"
                  onClick={() => {
                    handleUserClick(row._id, history);
                  }}
                >
                  {row.name} {row.surname}
                </Link>
              </TableCell>
              <TableCell className={classes.userEmailCell} align="center">
                {row.email}
              </TableCell>
              <TableCell className={classes.userStatusCell} align="center">
                <strong>
                  {displayStatus(row.status)}
                </strong>
              </TableCell>
              <TableCell className={classes.userControlCell} align="center">
                {actionCheckBox(row, classes, dispatch, history)}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      {displayNotifications()}
    </Paper>
  );
};

UsersContainer.propTypes = {
  classes: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    errorMessage: state.users.error,
    successMessage: state.users.success,
    messageConfirmed: state.users.confirmed
  };
}

export default connect(mapStateToProps, {
  getUsers,
  getSingleUser
})(withStyles(Container)(withRouter(UsersContainer)));