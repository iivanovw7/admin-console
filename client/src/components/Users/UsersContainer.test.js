import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import * as mocks from '../../__mocks__/';
import { getUsers } from '../../actions';
import { Container } from '../../components/UI/ThemeProperties';
import UsersContainer from '../../components/Users/UsersContainer';
import { history, storeFactory, limit, page } from '../../testUtils';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);

  const Composition = connect(null, { getUsers })(withStyles(Container, { withTheme: true })(withRouter(UsersContainer)));
  const props = {
    limit,
    page,
    search: mocks.USER.name,
    history: history,
    users: mocks.USERS.output
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders users container with styles,
   *  Checks if component rendered properly.
   */
  describe('Users container', () => {
    it('Should render HOC with styles properly', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} dispatch={mockFunc}/>
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();



