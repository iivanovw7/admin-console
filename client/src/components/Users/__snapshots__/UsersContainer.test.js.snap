// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users container Should render HOC with styles properly 1`] = `"<div class=\\"MuiPaper-root-59 MuiPaper-elevation2-63 MuiPaper-rounded-60 withRouter-UsersContainer--root-30 withRouter-Connect-WithStyles-withRouter-UsersContainer-----root-1\\"><table class=\\"MuiTable-root-86 withRouter-UsersContainer--tables-38 withRouter-Connect-WithStyles-withRouter-UsersContainer-----tables-9\\"><thead class=\\"MuiTableHead-root-87\\"><tr class=\\"MuiTableRow-root-88 MuiTableRow-head-91\\"><th class=\\"MuiTableCell-root-93 MuiTableCell-head-94 withRouter-UsersContainer--userNameCell-50 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userNameCell-21\\" scope=\\"col\\"><h3>Name</h3></th><th class=\\"MuiTableCell-root-93 MuiTableCell-head-94 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userEmailCell-51 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userEmailCell-22\\" scope=\\"col\\"><h3>Email</h3></th><th class=\\"MuiTableCell-root-93 MuiTableCell-head-94 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userStatusCell-52 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userStatusCell-23\\" scope=\\"col\\"><h3>Status</h3></th><th class=\\"MuiTableCell-root-93 MuiTableCell-head-94 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userControlCell-53 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userControlCell-24\\" scope=\\"col\\"><h3>Deactivate</h3></th></tr></thead><tbody class=\\"MuiTableBody-root-105\\"><tr class=\\"MuiTableRow-root-88\\"><th class=\\"MuiTableCell-root-93 MuiTableCell-body-95 withRouter-UsersContainer--userNameCell-50 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userNameCell-21\\" scope=\\"row\\"><button class=\\"MuiTypography-root-111 MuiTypography-body2-119 MuiTypography-colorPrimary-141 MuiTypography-inline-146 MuiLink-root-106 MuiLink-button-110 MuiLink-underlineHover-108\\">Admin ADMIN</button></th><td class=\\"MuiTableCell-root-93 MuiTableCell-body-95 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userEmailCell-51 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userEmailCell-22\\">admin@company.org</td><td class=\\"MuiTableCell-root-93 MuiTableCell-body-95 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userStatusCell-52 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userStatusCell-23\\"><strong>Active</strong></td><td class=\\"MuiTableCell-root-93 MuiTableCell-body-95 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userControlCell-53 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userControlCell-24\\"><span class=\\"MuiButtonBase-root-163 MuiIconButton-root-157 MuiPrivateSwitchBase-root-153 MuiCheckbox-root-147 MuiCheckbox-colorPrimary-151 MuiPrivateSwitchBase-checked-154 MuiCheckbox-checked-148 withRouter-UsersContainer--actionCheckBox-58 withRouter-Connect-WithStyles-withRouter-UsersContainer-----actionCheckBox-29\\"><span class=\\"MuiIconButton-label-162\\"><svg class=\\"MuiSvgIcon-root-166\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-156\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\" checked=\\"\\"></span><span class=\\"MuiTouchRipple-root-175\\"></span></span></td></tr><tr class=\\"MuiTableRow-root-88\\"><th class=\\"MuiTableCell-root-93 MuiTableCell-body-95 withRouter-UsersContainer--userNameCell-50 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userNameCell-21\\" scope=\\"row\\"><button class=\\"MuiTypography-root-111 MuiTypography-body2-119 MuiTypography-colorPrimary-141 MuiTypography-inline-146 MuiLink-root-106 MuiLink-button-110 MuiLink-underlineHover-108\\">Holt Aasaf</button></th><td class=\\"MuiTableCell-root-93 MuiTableCell-body-95 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userEmailCell-51 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userEmailCell-22\\">haasaf4q@tamu.edu</td><td class=\\"MuiTableCell-root-93 MuiTableCell-body-95 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userStatusCell-52 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userStatusCell-23\\"><strong>Active</strong></td><td class=\\"MuiTableCell-root-93 MuiTableCell-body-95 MuiTableCell-alignCenter-102 withRouter-UsersContainer--userControlCell-53 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userControlCell-24\\"><span class=\\"MuiButtonBase-root-163 MuiIconButton-root-157 MuiPrivateSwitchBase-root-153 MuiCheckbox-root-147 MuiCheckbox-colorPrimary-151 MuiPrivateSwitchBase-checked-154 MuiCheckbox-checked-148 withRouter-UsersContainer--actionCheckBox-58 withRouter-Connect-WithStyles-withRouter-UsersContainer-----actionCheckBox-29\\"><span class=\\"MuiIconButton-label-162\\"><svg class=\\"MuiSvgIcon-root-166\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-156\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\" checked=\\"\\"></span><span class=\\"MuiTouchRipple-root-175\\"></span></span></td></tr></tbody></table></div>"`;
